engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeAsyncCommandsTest.scala

```diff
@@ -0,0 +1,245 @@
+package org.enso.interpreter.test.instrument
+
+import org.enso.distribution.FileSystem
+import org.enso.distribution.locking.ThreadSafeFileLockManager
+import org.enso.interpreter.instrument.execution.Timer
+import org.enso.interpreter.test.Metadata
+import org.enso.pkg.{Package, PackageManager}
+import org.enso.polyglot._
+import org.enso.polyglot.runtime.Runtime.Api
+import org.enso.text.{ContentVersion, Sha3_224VersionCalculator}
+import org.graalvm.polyglot.Context
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.io.{ByteArrayOutputStream, File}
+import java.nio.file.{Files, Path, Paths}
+import java.util.UUID
+
+@scala.annotation.nowarn("msg=multiarg infix syntax")
+class RuntimeAsyncCommandsTest
+    extends AnyFlatSpec
+    with Matchers
+    with BeforeAndAfterEach {
+
+  // === Test Timer ===========================================================
+
+  class TestTimer extends Timer {
+    override def getTime(): Long = 0
+  }
+
+  // === Test Utilities =======================================================
+
+  var context: TestContext = _
+
+  class TestContext(packageName: String) extends InstrumentTestContext {
+    val tmpDir: Path = Files.createTempDirectory("enso-test-packages")
+    sys.addShutdownHook(FileSystem.removeDirectoryIfExists(tmpDir))
+    val lockManager = new ThreadSafeFileLockManager(tmpDir.resolve("locks"))
+    val runtimeServerEmulator =
+      new RuntimeServerEmulator(messageQueue, lockManager)
+
+    val pkg: Package[File] =
+      PackageManager.Default.create(tmpDir.toFile, packageName, "Enso_Test")
+    val out: ByteArrayOutputStream = new ByteArrayOutputStream()
+    val executionContext = new PolyglotContext(
+      Context
+        .newBuilder(LanguageInfo.ID)
+        .allowExperimentalOptions(true)
+        .allowAllAccess(true)
+        .option(RuntimeOptions.PROJECT_ROOT, pkg.root.getAbsolutePath)
+        .option(RuntimeOptions.LOG_LEVEL, "WARNING")
+        .option(
+          RuntimeOptions.INTERPRETER_SEQUENTIAL_COMMAND_EXECUTION,
+          "false"
```

Comment:
Our runtime tests are sequential, i.e. all the commands sent to the engine are executed in sequence. It is a very useful property when testing the code because it prevents all kinds of race conditions.
Unforutnately, you cannot test the program interruption with the sequential flow because you need to send an interruption command while the previous command (execute program) is still in progress. That's why there is a new suite.